File: 0739-daily-temperatures/0739-daily-temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function(temperatures) {
    let stack = [];
    let res = new Array(temperatures.length).fill(0);

    for(let i = 0; i < temperatures.length; i++){
        while(stack.length && temperatures[stack[stack.length - 1]] < temperatures[i]){
            let idx = stack.pop();
            res[idx] = i - idx;
        }
        stack.push(i);
    }
    return res;
};

File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temps) {
        Stack<Integer> stack = new Stack<>();
        int[] res = new int[temps.length];  

        for(int i = 0; i < temps.length; i++){
            while(!stack.isEmpty() && temps[stack.peek()] < temps[i]){
                int idx = stack.pop();
                res[idx] = i - idx;
            }
            stack.push(i);
        }

        return res;

    }
}